commands/rm.go

```diff
@@ -100,3 +127,56 @@ func rm(ctx context.Context, dockerCli command.Cli, ng *store.NodeGroup, keepSta
 	}
 	return err
 }
+
+func rmAllInactive(ctx context.Context, txn *store.Txn, dockerCli command.Cli, in rmOptions) error {
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	ll, err := txn.List()
+	if err != nil {
+		return err
+	}
+
+	builders := make([]*nginfo, len(ll))
+	for i, ng := range ll {
+		builders[i] = &nginfo{ng: ng}
+	}
+
+	eg, _ := errgroup.WithContext(ctx)
+	for _, b := range builders {
+		func(b *nginfo) {
+			eg.Go(func() error {
+				if err := loadNodeGroupData(ctx, dockerCli, b); err != nil {
+					return errors.Wrapf(err, "cannot load %s", b.ng.Name)
+				}
+				if b.ng.Dynamic {
+					return nil
+				}
+				var kn []store.Node
+				for idx, n := range b.ng.Nodes {
+					d := b.drivers[idx]
+					if d.info == nil || d.info.Status != driver.Running {
+						_ = d.di.Driver.Rm(ctx, true, !in.keepState, !in.keepDaemon)
+						continue
+					}
+					kn = append(kn, n)
+				}
+				b.ng.Nodes = kn
```

Comment:
I don't think this is the behavior we want. We shouldn't reconfigure the builders. If all the nodes are inactive then remove the builder. If at least one of them is running then skip it.

Reply:
ok I was not thinking like this for `rm --all-inactive`. for me it removes inactives builders and nodes so if a node is inactive in a builder it should be removed from this builder. This way we we only have active builders and nodes in our stack.

```
$ docker buildx ls
NAME/NODE     DRIVER/ENDPOINT                        STATUS   PLATFORMS
builder       docker-container
  builder0    unix:///var/run/docker.sock            inactive
  mac-mini-m1 tcp://sifive.home.com:2375      running  linux/rsicv64*, linux/riscv64
  sifive      tcp://mac-mini-m1.home.pl4gue.com:2376 Cannot connect to the Docker daemon at tcp://mac-mini-m1.home.com:2376. Is the docker daemon running?
builder2      docker-container
  builder20   unix:///var/run/docker.sock            inactive
desktop-linux                                        protocol not available
default *     docker
  default     default                                running                linux/amd64, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
```

here `builder0` and `sifive` will never be removed because `mac-mini-m1` is active. I think the user would want `builder0` and `sifive` removed because they are inactive. maybe we could allow that if `--force` is being used?

Reply:
We have `--leave` for that use case where you want to reconfigure builders. The use case for `rm --all-inactive` was that there are cases where you have too much stuff under `buildx ls` and want to do some housekeeping. Now you would need to write some loop that repeatedly calls `buildx rm <buildername>`. In practice it is even harder as you can't determine the ones you want to keep. So in reality it is a bunch of copy-paste. `--all-inactive` helps you with this loop and triggers `rm` for all filtered builders as if you invoked `rm` for all of them separately.

Removing a node from a builder doesn't make sense in here as this is not what `rm` does. Your case would be more like `--leave-all-inactive` but I don't think that case is something that users hit nearly as often and you will also not have tens of inactive nodes under a builder where you would need a loop to clear them.

Reply:
Ah yes completely forgot about the `--leave` flag!